app/models/reports/schema_v2.rb

```diff
@@ -226,5 +186,38 @@ def percentage(numerator, denominator)
       return 0 if numerator.nil? || denominator.nil? || denominator == 0 || numerator == 0
       ((numerator.to_f / denominator) * 100).round(PERCENTAGE_PRECISION)
     end
+
+    # Grab a particular summed field for a region. We return data in the format of:
+    #   { region_slug => { period_1 => x, period_2 => y }, ...}
+    # to maintain consistency w/ the format callers expect from the Repository.
+    #
+    # Note that we do filtering on the result set to limit the returned amount of data to the data that callers are
+    # requesting via the `periods` argument the Repository was created with.
+    def sum(region, field)
+      summed_field = "sum_#{field}"
+      facility_state_data(region)
+        .reject { |facility_state| Period.month(facility_state.month_date) < earliest_patient_recorded_at_period[region.slug] }
+        .select { |facility_state| facility_state.period_month_date.in?(periods) }
+        .to_h { |facility_state| [Period.month(facility_state.month_date), facility_state.public_send(summed_field)] }
+        .tap { |hsh| hsh.default = 0 }
+    end
+
+    delegate :sql, to: Arel
+
+    # Grab all the summed data for a particular region grouped by month_date.
+    # We need to use COALESCE to avoid getting nil back from some of the values, and we need to use
+    # `select` because the `sum` methods in ActiveRecord can't sum multiple fields.
+    # We also order by `month_date` because some code in the views expects elements to be ordered by Period from
+    # oldest to newest - it also makes reading output in specs and debugging much easier.
+    memoize def facility_state_data(region)
+      calculations = FIELDS.map { |field| Arel.sql("COALESCE(SUM(#{field}::int), 0) as sum_#{field}") }
+
+      FacilityState.for_region(region)
+        .where("cumulative_registrations IS NOT NULL OR cumulative_assigned_patients IS NOT NULL")
+        .order(:month_date)
+        .group(:month_date)
+        .select(:month_date)
+        .select(*calculations)
```

Comment:
Do we not want to filter to the given period here? As in, read only the months that the repository is initiated for?

Reply:
Nah, not yet, I want to compare some performance numbers and query plans w/ both approaches before adding that to the `where` clause.

Reply:
What's the trade-off though? As in, what do we lose in adding a where clause given that the repo is instantiated for a given period anyway?